### PR TITLE
Removes empty catch blocks

### PR DESCRIPTION
**Pull request by fandy**
Fixes https://github.com/Shopify/javascript/issues/109

/cc @lemonmade

### CODE REVIEW

File: packages/shopify-codemod/transforms/remove-empty-catch-blocks.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+
+  return j(source)
+    .find(j.TryStatement, {
+      type: 'TryStatement',
+      handler: {
+        type: 'CatchClause',
+      },
+    })
+      .forEach(({node}) => {
+        if (node.handler.body.body.length === 0) {
```
Comment (lemonmade) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: Let's move this check into the filter of `find` (`body: {body: (body) => body.length === 0}` in the `handler` clause.

File: packages/shopify-codemod/transforms/remove-empty-catch-blocks.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export default function removeEmptyCatchBlocks({source}, {jscodeshift: j}, {printOptions = {}}) {
+
+  return j(source)
+    .find(j.TryStatement, {
+      type: 'TryStatement',
+      handler: {
+        type: 'CatchClause',
```
Comment (lemonmade) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: Can you use `j.TryStatement.name`/ `j.CatchClause.name` instead of the strings? I know we are inconsistent about this but I'd like to stick to that style.

File: packages/shopify-codemod/transforms/remove-empty-catch-blocks.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+      },
+    })
+      .forEach(({node}) => {
```
Comment (lemonmade) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: Indentation a little weird here.

File: packages/shopify-codemod/transforms/remove-empty-catch-blocks.js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        if (node.handler.body.body.length === 0) {
+          delete node.handler;
```
Comment (lemonmade) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: Also, let's try this using the `NodePath` methods — `path.get('handler').replace()`

File: packages/shopify-codemod/test/fixtures/remove-empty-catch-blocks/basic.input.js [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+  foo();
+} catch (undefined) {
+
+} finally {
+  bar();
+}
+
+try {
+  foo();
+} catch (e) {} finally {
```
Comment (lemonmade) [inline packages/shopify-codemod/test/fixtures/remove-empty-catch-blocks/basic.input.js]: Please add a test case with a non-empty `catch` block to show they are not removed.

**Comment by lemonmade** [comment]
Also, remember docs + adding to the `esify` `lib/codemod` manifest.

**Comment by fandy** [comment]
Updated. :)

File: packages/shopify-codemod/transforms/remove-empty-catch-blocks.js (hunk 2 above)
Comment (fandy) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: In that case I should probably update the other transforms to use `j.Type.name` instead of literals.

Comment (lemonmade) [inline packages/shopify-codemod/transforms/remove-empty-catch-blocks.js]: There's a lot of those, we can do it later (we can even write a quick transform to do it for us :P)

**Comment by fandy** [comment]
@lemonmade is this PR ready to merge?

File: packages/shopify-codemod/test/fixtures/remove-empty-catch-blocks/basic.output.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+
+try {
+  foo();
+} finally {
+  bar();
+}
+
+try {
+  foo();
+} finally {}
```
Comment (lemonmade) [inline packages/shopify-codemod/test/fixtures/remove-empty-catch-blocks/basic.output.js]: We are missing a `try` with a `catch` but without a `finally` — The catch can't be removed in that case.

**Comment by lemonmade** [comment]
One minor comment, otherwise LGTM 👍

**Comment by GoodForOneFare** [comment]
Fixed in decaf.

**Comment by fandy** [comment]
In that case I'll close this branch.
